File: backend/SafetyAndHealth.Db/Models/UserCertificate.cs
﻿using System;
namespace SafetyAndHealth.Db.Models
{
    public class UserCertificate : RelationalEntity
    {
        public string Number { get; set; }

        public DateTime StartDate { get; set; }

        public long? Group { get; set; }

        public User User { get; set; }

        public string UserId { get; set; }

        public FileDescription File { get; set; }

        public long? FileId { get; set; }

        public Certificate Certificate { get; set; }

        public long CertificateId { get; set; }
    }
}

File: frontend/src/api/models/UserCertificate.ts
import { FileDescriptionDto } from "./FileDescription";
import { BaseDtoEntity } from "./BaseModel";
import { CertificateDto } from "./Certificate";

export interface UserCertificateDto extends BaseDtoEntity {
    number: string;

    startDate: Date;

    group?: number;

    userId: string;

    file: FileDescriptionDto;

    certificate: CertificateDto;

    expiresDate?: Date;
}

export interface UserCertificateCreateUpdateDto extends BaseDtoEntity {
    number: string;

    startDate: Date | string;

    group: number | null;

    userId: string;

    certificateId: number;
}